### PR TITLE
Experiment page

### PR DESCRIPTION
**Pull request by nendow02**
Added experiment page with full functionality.

### CODE REVIEW

**Review by archishadatta — APPROVED** [review]
Just pushed any css fixes for the experiment page, should be good now.

File: src/components/pages/experiment.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
+import '../../styles/experiment.scss';
+import { useState, useEffect } from 'react';
+import { BooleanDropdown } from '../shared/booleanDropdown';
+import {lineOfCode, CodeFormat}  from  '../shared/codeFormat';
+import {Color} from '../shared/constants';
```
Comment (trevorong) [inline src/components/pages/experiment.tsx:5]:
```suggestion
import { lineOfCode, CodeFormat }  from  '../shared/codeFormat';
import { Color } from '../shared/constants';
```

File: src/components/shared/generalDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export function GeneralDropdown(props:GeneralDropdownProps):JSX.Element  {
+  const options: {value:string, label:string, className:string}[] = [];
+  props.options.map((option) => {
+    if (option != props.options[props.options.length-1]) {
+      options.push({value: option, label: option, className: 'option'});
+    }
+    else {
+      options.push({value: option, label: option, className: 'option last'});
+    }
+  });
```
Comment (trevorong) [inline src/components/shared/generalDropdown.tsx]:
```suggestion
  const options: {value:string, label:string, className:string}[] = props.options.map((option) => {
    const opt = {value: option, label: option, className: 'option'};
    if (option === props.options[props.options.length-1]) {
      opt.className = 'option last';
    } 
    return opt;
  });
```
options.map should be moved into the initialization step (better not to have function calls inside JSX element body)

**Review by trevorong — COMMENTED** [review]
1 small change but otherwise LGTM

**Comment by reginawang3495** [comment]
@trevorong is helping me review (thanks trevor~)!! his approval is mine also :))

**Pull request merged**